--- contrib/coraza-spoa.service	2025-09-11 21:23:47.519489669 +0200
+++ contrib/coraza-spoa.service	2025-09-11 21:23:47.520489678 +0200
@@ -3,7 +3,7 @@
 Documentation=https://www.coraza.io
 
 [Service]
-ExecStart=/usr/bin/coraza-spoa -config=/etc/coraza-spoa/config.yaml
+ExecStart=/usr/sbin/coraza-spoa -config=/etc/coraza-spoa/coraza-spoa.yaml
 WorkingDirectory=/
 Restart=always
 Type=exec
@@ -45,7 +45,7 @@
 InaccessiblePaths=-/srv
 #InaccessiblePaths=-/bin
 InaccessiblePaths=-/bin/bash
-inaccessiblepaths=-/bin/find
+Inaccessiblepaths=-/bin/find
 InaccessiblePaths=-/bin/less
 InaccessiblePaths=-/bin/zcat
 InaccessiblePaths=-/bin/rm
